Makefile

```diff
@@ -12,130 +12,219 @@
+	go test -v -mod=readonly -cover -race -timeout 10s  ./pkg/eniconfig/...
+	go test -v -mod=readonly -cover -race -timeout 10s  ./pkg/ipamd/...
+
+# Build the unit test driver container image.
 build-docker-test:
-	@docker build -f scripts/dockerfiles/Dockerfile.test -t amazon-k8s-cni-test:latest .
+	docker build $(DOCKER_BUILD_FLAGS) \
+		-f scripts/dockerfiles/Dockerfile.test \
+		-t $(TEST_IMAGE_NAME) \
+		.
 
+# Run unit tests inside of the testing container image.
 docker-unit-test: build-docker-test
-	docker run -e GO111MODULE=on \
-		amazon-k8s-cni-test:latest make unit-test
+	docker run $(DOCKER_RUN_ARGS) \
+		$(TEST_IMAGE_NAME) \
+		make unit-test
 
-# Build metrics
+# Build metrics helper agent.
 build-metrics:
-	GOOS=linux GOARCH=$(ARCH) CGO_ENABLED=0 go build -ldflags="-s -w" -o cni-metrics-helper ./cmd/cni-metrics-helper/
+	go build -mod=readonly -ldflags="-s -w" -o cni-metrics-helper ./cmd/cni-metrics-helper
 
-# Build metrics Docker image
+# Build metrics helper agent Docker image.
 docker-metrics:
-	@docker build --build-arg arch="$(ARCH)" -f scripts/dockerfiles/Dockerfile.metrics -t "amazon/cni-metrics-helper:$(VERSION)" .
+	docker build $(DOCKER_BUILD_FLAGS) \
+		-f scripts/dockerfiles/Dockerfile.metrics \
+		-t "$(METRICS_IMAGE_NAME)" \
+		.
 	@echo "Built Docker image \"amazon/cni-metrics-helper:$(VERSION)\""
 
+# Run metrics helper unit test suite (must be run natively).
+metrics-unit-test: CGO_ENABLED=1
+metrics-unit-test: GOARCH=
 metrics-unit-test:
-	GOOS=linux CGO_ENABLED=1 go test -v -cover -race -timeout 10s ./cmd/cni-metrics-helper/metrics/...
+	go test -v -cover -race -timeout 10s -mod=readonly \
+		./cmd/cni-metrics-helper/metrics/...
 
+# Run metrics helper unit test suite in a container.
 docker-metrics-test:
-	docker run -v $(shell pwd):/usr/src/app/src/github.com/aws/amazon-vpc-cni-k8s \
-		--workdir=/usr/src/app/src/github.com/aws/amazon-vpc-cni-k8s \
-		--env GOPATH=/usr/src/app \
-		golang:1.10 make metrics-unit-test
-
-# Build both CNI and metrics helper
-all: docker docker-metrics
+	docker run $(DOCKER_RUN_FLAGS) \
+		-v $(shell pwd -P):/src --workdir=/src \
+		-e GOARCH -e GOOS -e GO111MODULES \
+		$(GOLANG_IMAGE) \
+		make metrics-unit-test
 
+# Generate descriptors for supported ENI configurations.
 generate-limits:
-	go run pkg/awsutils/gen_vpc_ip_limits.go
+	go run -mod=readonly pkg/awsutils/gen_vpc_ip_limits.go
 
-# golint
-# To install: go get -u golang.org/x/lint/golint
+# Download from upstream
+#
+# TODO: confirm this is still a used entrypoint or remove.
+download-portmap: portmap
+
+# Fetch portmap the port-forwarding management CNI plugin
+portmap: FETCH_VERSION=0.7.5
+portmap: FETCH_URL=https://github.com/containernetworking/plugins/releases/download/v$(FETCH_VERSION)/cni-plugins-$(GOARCH)-v$(FETCH_VERSION).tgz
+portmap: VISIT_URL=https://github.com/containernetworking/plugins/tree/v$(FETCH_VERSION)/plugins/meta/portmap
+portmap:
```

Comment:
In my experience, it's more common to see this sort of target written as `make portmap` (and [I'd like to remove `download-portmap`](https://github.com/aws/amazon-vpc-cni-k8s/pull/837/files/ebf5b38a81d45c5300347357ee1a733cb9c8ca25#diff-b67911656ef5d18c4ae36cb6741b7965R170)), the `download-portmap` target *does tell you what it does* but its more conventional to declare what it is that you "want" to make in the context of `make`. The act of downloading the executable is besides the declaration that `portmap` *should be made to be on disk.. somehow*.

To be perfectly clear, this is definitely not how it has to be and can certainly be renamed to `download-portmap`. The current arrangement keeps compatibility with the prior usage (calling `make download-portmap`, and only downloading if the executable is missing) so that any unseen callers will continue working. I did this to keep from messing up any system that I'm unaware of.

I'd (personally) prefer to see this written as `make portmap` to align with the declarative nature of make but its not my call to.. `make`! Let me know what you think and what considerations ought to be made to keep existing callers working :+1:

Reply:
`make portmap` sounds good, feel free to change to that.